### ISSUE TITLE
Support remote jupyter server connections to same degree as webviews 

### ISSUE THREAD

**Issue by DonJayamanne** [microsoft/vscode-jupyter#4456]
Ensure Remote in native notebooks work the same was as in WebViews.
Created a separate issue as https://github.com/microsoft/vscode-jupyter/issues/254 was meant for a new way of impelenting remote for Native Notebooks.

**Comment by DonJayamanne** [comment]
### Parity UI/UX experience

Prototype Link (all users with MS SSO should be able to sign in): https://www.figma.com/file/OGTDR9c9U5sFemy6xMlaE2/Visual-Studio-Code-Toolkit-Community?node-id=91%3A627

**Screen: Local**
![Local](https://user-images.githubusercontent.com/10951869/105392940-a2590400-5bd0-11eb-91b5-0b4425a3f13e.png)
**Screen: Local / Connections**
![Local _ Connections](https://user-images.githubusercontent.com/10951869/105392947-a38a3100-5bd0-11eb-9786-6a41abfe4f5d.png)
**Screen: Local / Connections / URI**
![Local _ Connections _ URI](https://user-images.githubusercontent.com/10951869/105392951-a4bb5e00-5bd0-11eb-8b63-5d54f897596c.png)
**Screen: Remote**
![Remote](https://user-images.githubusercontent.com/10951869/105392965-a84ee500-5bd0-11eb-96e0-c4940bc18849.png)
**Screen: Remote/Connections**
![Remote _ Connections](https://user-images.githubusercontent.com/10951869/105392969-a9801200-5bd0-11eb-8c3c-200b7fd4c8f7.png)

**Prototype in Image Form**
**Local Flow:**
![image](https://user-images.githubusercontent.com/10951869/105393156-ddf3ce00-5bd0-11eb-91d0-1e7c736fb306.png)
**Remote Flow:**
![image](https://user-images.githubusercontent.com/10951869/105393217-f237cb00-5bd0-11eb-9b8c-2b4b9c98b109.png)

**Comment by rchiodo** [comment]
This doesn't seem to be working? I get a couple of problems

1) After picking the remote server, I reload VS code. Upon opening a notebook the pick a server dialog comes up again.
1) There are no kernels listed

**Comment by DonJayamanne** [comment]
Blocked on https://github.com/microsoft/vscode/issues/114955

**Comment by DavidKutu** [comment]
Works for me
Validated
